docs/DeveloperGuide.md

```diff
@@ -0,0 +1,123 @@
+# RepoSense - Developer Guide
+Thank you for your interest in contributing to RepoSense!
+- [Setting up](#setting-up)
+  - [Prerequisites](#prerequisites)
+  - [Setting up the project in your computer using IntelliJ](#setting-up-the-project-in-your-computer-using-intellij)
+  - [Verifying the setup](#verifying-the-setup)
+  - [Configuring the coding style](#configuring-the-coding-style)
+- [Architecture](#architecture)
+  - [Parser](#parser)
+  - [Git](#git)
+  - [CommitsReporter](#commitsreporter)
+  - [AuthorshipReporter](#authorshipreporter)
+  - [ReportGenerator](#reportgenerator)
+  - [System](#system)
+  - [Model](#model)
+- [HTML Dashboard](#html-dashboard)
+
+## Setting up
+
+### Prerequisites
+1. **JDK `1.8.0_60`** or later.
+2. **git** on the command line.
+3. **findstr** for *Windows*, **grep** for *macOS* or *Linux* on the command line.
+ > Check that the tools exist on your OS terminal by typing its name on the terminal and ensure that the terminal does not output messages such as `not found` or `not recognized`.
```

Comment:
Minor: If i run just `findstr` in a Windows terminal, I get a `FINDSTR: bad command line` error. This is actually an error from FINDSTR but someone might mistake it as the tool itself is missing.

Reply:
Also, if these two are supposed to be available in the respective OS by default, may be no need to mention? The app can spit out an explanatory error message if the tool is not found or a rare case.